service/src/test/java/bio/terra/workspace/service/workspace/flight/DeleteAzureContextFlightTest.java

```diff
@@ -0,0 +1,356 @@
+import bio.terra.workspace.service.resource.controlled.model.ControlledResourceFields;
+import bio.terra.workspace.service.resource.controlled.model.ManagedByType;
+import bio.terra.workspace.service.resource.exception.ResourceNotFoundException;
+import bio.terra.workspace.service.resource.model.CloningInstructions;
+import bio.terra.workspace.service.spendprofile.SpendConnectedTestUtils;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.flight.create.azure.CreateAzureContextFlight;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DeleteAzureContextFlightTest extends BaseAzureTest {
+  /**
+   * How long to wait for a delete context Stairway flight to complete before timing out the test.
+   */
+  private static final Duration DELETION_FLIGHT_TIMEOUT = Duration.ofMinutes(3);
+  /**
+   * How long to wait for a create context Stairway flight to complete before timing out the test.
+   */
+  private static final Duration CREATION_FLIGHT_TIMEOUT = Duration.ofMinutes(3);
+
+  @Autowired private WorkspaceService workspaceService;
+  @Autowired private ControlledResourceService controlledResourceService;
+  @Autowired private SpendConnectedTestUtils spendUtils;
+  @Autowired private UserAccessUtils userAccessUtils;
+  @Autowired private JobService jobService;
+  @Autowired private WorkspaceConnectedTestUtils testUtils;
+  @Autowired private AzureTestUtils azureTestUtils;
+
+  private UUID workspaceUuid;
+
+  @BeforeEach
+  public void setup() {
+    // Create a new workspace at the start of each test.
+    UUID uuid = UUID.randomUUID();
+    Workspace request =
+        Workspace.builder()
+            .workspaceId(uuid)
+            .userFacingId("a" + uuid.toString())
+            .workspaceStage(WorkspaceStage.MC_WORKSPACE)
+            .spendProfileId(spendUtils.defaultSpendId())
+            .build();
+    workspaceUuid =
+        workspaceService.createWorkspace(request, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  @AfterEach
+  public void tearDown() {
+    workspaceService.deleteWorkspace(workspaceUuid, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  @Test
+  void deleteContextDo() throws Exception {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    FlightMap createParameters =
+        azureTestUtils.createAzureContextInputParameters(workspaceUuid, userRequest);
+
+    // Create the azure context.
```

Comment:
Nit: the different test cases have blocks of common setup that could be put into helper methods (creating context, creating Azure IP resource).

Reply:
Thanks for doing this cleanup!